lib/server/sse_handler.dart

```diff
@@ -55,9 +76,35 @@ class SseConnection extends StreamChannelMixin<String> {
   @override
   Stream<String> get stream => _incomingController.stream;
 
+  void _acceptReconnection(Sink sink) {
+    _isTimingOut = false;
+    _sink = sink;
+    _outgoingStreamSubscription.resume();
+  }
+
+  void _handleDisconnect() {
+    if (_keepAlive == null) {
+      _close();
+    } else if (!_isTimingOut) {
+      _outgoingStreamSubscription.pause();
+      _isTimingOut = true;
+      // If after the timeout period we're still in this state, we'll close.
+      Timer(_keepAlive, () {
+        if (_isTimingOut) {
+          _isTimingOut = false;
+          _close();
+        }
+      });
+    }
+  }
+
+  // TODO(dantup): @visibleForTesting?
+  void closeSink() => _sink.close();
```

Comment:
There is a slightly better pattern to this but it will require some work.

Let's move this `sse_handler.dart` under `src`. Make this `closeSink` a top level method that takes an `SseConnection`. Now export everything under `sse_handler.dart` but  `closeSink`.

In the test you can import from `src` without concern and call `closeSink`. Note you can also annotate that it is only visible for testing.

See this parallel example:
https://github.com/dart-lang/webdev/blob/2958ede70f20c402e869b4169f3d2f97162e51d4/dwds/lib/src/connections/debug_connection.dart#L50

Reply:
Ah, nice! I've done that - and I committed the move as its own changeset to try and avoid Git just showing "this file deleted" and "this file added", however the overall diff here seems to have ignored that. This could cause conflicts if anyone else has outstanding changes in that file.

Reply:
I don't think we have any pending PRs so LGTM.